crates/hir-def/src/adt.rs

```diff
@@ -143,43 +145,87 @@ fn parse_repr_tt(tt: &Subtree) -> Option<ReprData> {
 
 impl StructData {
     pub(crate) fn struct_data_query(db: &dyn DefDatabase, id: StructId) -> Arc<StructData> {
+        db.struct_data_with_diagnostics(id).0
+    }
+
+    pub(crate) fn struct_data_with_diagnostics_query(
+        db: &dyn DefDatabase,
+        id: StructId,
+    ) -> (Arc<StructData>, Arc<Vec<DefDiagnostic>>) {
```

Comment:
```suggestion
    ) -> (Arc<StructData>, Arc<[DefDiagnostic]>) {
```
No need for a `Vec` here, `Arc` can already hold a slice. We can just `info` the vec into an arc at the end